homeassistant/components/brother/config_flow.py

```diff
@@ -64,6 +69,61 @@ class BrotherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             step_id="user", data_schema=DATA_SCHEMA, errors=errors
         )
 
+    async def async_step_zeroconf(self, user_input=None):
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
+
+        try:
+            if not user_input.get("name").startswith("Brother"):
+                return self.async_abort(reason="not_brother_printer")
+        except AttributeError:
+            return self.async_abort(reason="not_brother_printer")
+
+        # Hostname is format: brother.local.
+        self.host = user_input["hostname"].rstrip(".")
+
+        try:
+            self.brother = Brother(self.host)
+            await self.brother.async_update()
+        except (ConnectionError, SnmpError, UnsupportedModel):
+            return self.async_abort(reason="connection_error")
+
+        # Check if already configured
+        await self.async_set_unique_id(self.brother.serial.lower())
+        self._abort_if_unique_id_configured()
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update(
```

Comment:
Why do we need to update context here?

Reply:
We need update context to fill `flow_title` string. Or maybe there is another way?

![brother](https://user-images.githubusercontent.com/478555/72895755-6faad000-3d1e-11ea-98ec-756b34857833.png)

Reply:
I'm not sure about this. It seems we're doing it like this in the integrations I've checked.

Reply:
@balloob?

Reply:
Yeah that's correct. If we have placeholders in the title we need to add that to the context.